### DISCUSSION TITLE
question regarding tensors:

### DISCUSSION THREAD

**Discussion by alikazim1** [mrdbourke/pytorch-deep-learning#801] [Q&A]
In the following tensor, why the dim=1 corrosponds to 3 in torch.Size([1,3,3]). Since there is only one bracket at the second position it should be 1. ? why? please guide me on this. Thanks...
<img width="871" alt="Screenshot 2024-01-11 202726" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/115345833/1e9c20e2-5822-4847-b5e2-eb29e9fcc891">
<img width="871" alt="Screenshot 2024-01-11 202726" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/115345833/590e6853-6534-4078-be6f-918330048024">

**Comment by LivelyAndrew** [comment]
this is how i understand it. dim=1 has 3 brackets:
BRACKET1 = [1, 2, 3] BRACKET2= [3, 6, 9] and BRACKET3=[2, 4, 5]

dim0 = [[1, 2, 3], [3, 6, 9], [2, 4, 5]] = 1 bracket (the biggest and outermost bracket)
dim2 = the count of numbers in the innermost bracket (each bracket in dim =1 has 3 numbers in them)
[1, 2, 3] = has 3 
[3, 6, 9] = has 3
[2, 4, 5] = has 3

![modified screenshot](https://github.com/mrdbourke/pytorch-deep-learning/assets/70732859/6955c2e0-48aa-4485-855d-98b4939a925b)
